addons/purchase/models/purchase_order_line.py

```diff
@@ -558,9 +576,12 @@ def _prepare_account_move_line(self, move=False):
             'price_unit': self.currency_id._convert(self.price_unit, aml_currency, self.company_id, date, round=False),
             'tax_ids': [(6, 0, self.taxes_id.ids)],
             'purchase_line_id': self.id,
```

Comment:
A lot of the things set here seem to be common between sale and purchase ; why not setting them in the mixin, and only add the required stuff in each of so/po lines ?